File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.py
class Solution(object):
    def getSmallestString(self, n, k):
        ans = ''

        while n != 0:
            maxAfterN = 26 * (n-1)

            if maxAfterN >= k:
                ans += 'a'
                k -= 1
            elif maxAfterN < k:
                ans += chr(k - maxAfterN + 96)
                k = maxAfterN

            n -= 1
        
        return ans

File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.swift
class Solution {
    func getSmallestString(_ n: Int, _ k: Int) -> String {
        var ans: String = ""
        var kk: Int = k
        
        for idx in (0..<n).reversed() {
            let maxAfterN: Int = 26 * idx
            
            if maxAfterN >= kk {
                ans += "a"
                kk -= 1
            } else if maxAfterN < kk {
                let value = kk - maxAfterN + 96
                
                ans += String(UnicodeScalar(value)!)
                kk = maxAfterN
            }
        }
        
        return ans
    }
}


